src/pages/AddNewBooksForm/BookForm/BookForm.tsx

```diff
@@ -95,69 +104,102 @@ const BookForm: React.FC<BookFormProps> = ({
       hasErrors = true
     }
 
-    if (amount <= 0) {
+    if (numberOfAvailableCopies <= 0) {
       setAmountError(true)
       hasErrors = true
     }
 
     return hasErrors
   }
 
+  const resetState = () => {
+    setBookCategory([])
+    setTitle('')
+    setAuthors([])
+    setNumberOfAvailableCopies(1)
+    setDescription('')
+    setImageUrl('')
+    setResetTrigger(prev => prev + 1)
+  }
+
   const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault()
     const hasErrors = validateInputs()
     if (hasErrors) return
 
-    const newBook = {
-      title,
-      author,
-      categories: bookCategory,
-      amount,
-      description,
-      imageUrl,
+    if (!addedBooks) {
+      console.error('No books to submit!')
+      return
     }
-    resetState()
 
-    console.log(newBook)
-    console.log(addedBooks)
+    const updatedBooks = addedBooks.map(bookWithFile => {
+      const bookCategoryToSnakeCase = categoryToSnakeCase(
+        bookWithFile.book.bookCategories.map(category => category.toUpperCase())
+      )
+
+      return {
+        file: bookWithFile.file,
+        book: {
+          ...bookWithFile.book,
+          bookCategories: bookCategoryToSnakeCase,
+        },
+      }
+    })
 
-    fetch('/add-book', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
+    const newBook: BookWithFile = {
+      book: {
+        title,
+        authors,
+        bookCategories: categoryToSnakeCase(
+          bookCategory.map(category => category.toUpperCase())
+        ),
+        numberOfAvailableCopies,
+        description,
+        imageUrl,
       },
-      body: JSON.stringify(newBook),
-    }).then(response => response.json())
-  }
+      file: imageFile,
+    }
 
-  const resetState = () => {
-    setBookCategory([])
-    setTitle('')
-    setAuthor('')
-    setAmount(1)
-    setDescription('')
-    setImageUrl('')
-    setResetTrigger(prev => prev + 1)
+    console.log(newBook.file)
+
+    resetState()
+
+    const booksToAdd = [...updatedBooks, newBook]
+
+    apiService
+      .addBooks(booksToAdd)
+      .then(results => {
+        console.log('Books added successfully:', results)
+        if (setAddedBooks != undefined) setAddedBooks([])
+      })
+      .catch(error => {
+        console.error('Error adding books:', error)
```

Comment:
While we're handling errors, we can add a [snackbar](https://mui.com/material-ui/react-snackbar/) to also tell the user that there is an error. Here the snackbar can just say `Error adding books`, without error details